esphome/components/esp32/__init__.py

```diff
@@ -295,6 +296,8 @@ async def to_code(config):
     conf = config[CONF_FRAMEWORK]
     cg.add_platformio_option("platform", conf[CONF_PLATFORM_VERSION])
 
+    cg.add_platformio_option("extra_scripts", "post:post_build.py")
```

Comment:
Let's make `extra_scripts` a list, so that it's merged with user-supplied values.
```suggestion
    cg.add_platformio_option("extra_scripts", ["post:post_build.py"])
```